spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -409,13 +435,14 @@ object DistriOptimizer {
           recordsProcessedThisEpoch = 0
         }
 
-        optimMethod.state.update("recordsProcessedThisEpoch", recordsProcessedThisEpoch)
-
-        optimMethod.state.update("epoch", driverState[Int]("epoch"))
-        optimMethod.state.update("neval", driverState[Int]("neval"))
-        optimMethod.state.update("Loss", driverState[Float]("Loss"))
-        if (validationMethods.isDefined) {
-          optimMethod.state.update("score", driverState[Float]("score"))
+        optimMethods.map { case (moduleName, optimMethod) =>
+          optimMethod.state.update("recordsProcessedThisEpoch", recordsProcessedThisEpoch)
```

Comment:
Can optimMethods share the same state ?

Reply:
We can't, the state also saves some internal buffer, the buffer will conflict if we share states.